bootstrap/api/swagger.yaml

```diff
@@ -0,0 +1,133 @@
+swagger: "2.0"
+info:
+  description: "Deployment API."
+  version: "1.0.0"
+  title: "Kubeflow Deployment"
+  termsOfService: "https://www.kubeflow.org/"
+  contact:
+    email: "kubeflow-engineering@google.com"
+  license:
+    name: "Apache 2.0"
+    url: "http://www.apache.org/licenses/LICENSE-2.0.html"
+host: "https://www.kubeflow.org/"
+basePath: "/kfctl"
+tags:
+- name: "deployment"
+  description: "A kubeflow deployment on GCP"
+  externalDocs:
+    description: "Find out more"
+    url: "https://www.kubeflow.org/docs/started/getting-started-gke/"
+schemes:
+- "https"
+paths:
+  /e2eDeploy:
+    post:
+      tags:
+      - "deployment"
+      summary: "Create a new kubeflow deployment on GCP"
+      description: ""
+      operationId: "createDeployment"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+      - in: "body"
+        name: "body"
+        description: "Pet object that needs to be added to the store"
+        required: true
+        schema:
+          $ref: "#/definitions/CreateDeployment"
```

Comment:
Following K8s and GCP style can the body of the Create request just be the spec for the deployment? 
i.e. a Create and Get request use the same object just with different fields.

Reply:
Perhaps instead of Create we should just have an Update method which would create it if it doesn't exist?